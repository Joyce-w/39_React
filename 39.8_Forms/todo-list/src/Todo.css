.Todo {
    width: 10em;
}

.Todo-btn{
    border: none;
    border-radius: 14px;
    margin: 0.5em;
    background-color: aliceblue;
}
