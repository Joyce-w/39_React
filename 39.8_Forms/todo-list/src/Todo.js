import React from "react";
import './Todo.css'


const Todo = ({ id, task , remove}) => {
    return (
            <div data-testid="Todo-test" id={ id } className="Todo">
            <li className="Todo-li">
                {task} 
                <button className="Todo-btn" onClick={ remove }>x</button>
            </li>
            </div>
    )
}

export default Todo;