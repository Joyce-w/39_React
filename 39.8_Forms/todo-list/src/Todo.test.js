import react from "react"
import { render } from '@testing-library/react';
import Todo from "./Todo"

//smoketest
test('renders learn react link', () => {
  render(<Todo />);
});

//snapshot
it("matches snapshot", () => {
  const { asFragment } = render(<Todo />)
  expect(asFragment()).toMatchSnapshot();
})