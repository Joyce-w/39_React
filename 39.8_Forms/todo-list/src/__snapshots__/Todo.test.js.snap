// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
    data-testid="Todo-test"
  >
    <li
      class="Todo-li"
    >
      <button
        class="Todo-btn"
      >
        x
      </button>
    </li>
  </div>
</DocumentFragment>
`;
